src/modules/sidebar/sidebar.component.ts

```diff
@@ -1,13 +1,34 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
 
-import { StacheNav, StacheNavLink } from '../nav';
+import { StacheNav, StacheNavLink, StacheNavService } from '../nav';
 
 @Component({
   selector: 'stache-sidebar',
   templateUrl: './sidebar.component.html',
   styleUrls: ['./sidebar.component.scss']
 })
-export class StacheSidebarComponent implements StacheNav {
+export class StacheSidebarComponent implements StacheNav, OnInit {
   @Input()
   public routes: StacheNavLink[];
+
+  public constructor(
+    private navService: StacheNavService) { }
+
+  public ngOnInit(): void {
+    if (!this.routes) {
+      const activeRoutes = this.navService.getActiveRoutes();
+      this.routes = this.filterRoutes(activeRoutes);
+    }
+  }
+
+  private filterRoutes(activeRoutes: StacheNavLink[]): StacheNavLink[] {
+    const root = activeRoutes[0];
+
+    root.children.unshift({
```

Comment:
Hmm. Yeah, I still don't love it, but we can see how it goes. I wish I could test it locally and see how it actually renders...